File: dataset/human/c/file_1839.c
#include <stdio.h>

int main(void){
	int n;
	scanf("%d", &n );
	int a[n], b[n], c[n-1];
  int ans=0;
	for(int i=0;i<n;i++){
		scanf("%d", &a[i]);
	}
	  for(int i=0;i<n;i++){
		scanf("%d", &b[i]);
      	ans += b[i];
	}
  for(int i=0;i<n-1;i++){
		scanf("%d", &c[i]);
	}

	for(int i=1;i<n;i++){
      if(a[i]-a[i-1]==1)	ans+=c[a[i-1]-1];
    }
	printf("%d", ans);

	return 0;
	
}

File: dataset/human/python/file_1839.py
import itertools

def solve(n, m, a, b):
    for i, j in itertools.product(range(n - m + 1), repeat=2):
        if all(a[i + x][j + y] == b[x][y] for x, y in itertools.product(range(m), repeat=2)): return "Yes"
    return "No"


if __name__ == '__main__':
    n, m = map(int, input().split())
    a = [[token for token in input()] for _ in range(n)]
    b = [[token for token in input()] for _ in range(m)]

    ans = solve(n, m, a, b)
    print(ans)
